src/software/ai/hl/stp/stp.cpp

```diff
@@ -137,17 +145,58 @@ std::vector<std::shared_ptr<Tactic>> STP::assignRobotsToTactics(
     // https://github.com/saebyn/munkres-cpp is the implementation of the Hungarian
     // algorithm that we use here
 
-    if (world.friendlyTeam().numRobots() < tactics.size())
+    auto friendly_team         = world.friendlyTeam();
+    auto& friendly_team_robots = friendly_team.getAllRobots();
+
+    // Special handling for the Goalie, since there should only ever be one and only
+    // one robot per team is ever permitted to be one
```

Comment:
nit: 
```suggestion
    // one robot assigned per team
```

Reply:
Changed to "    // one robot per team is assigned to be one", is that OK?

Reply:
It just needs to make sense with the previous line

Reply:
Rephrased again.